logging/internal/common.go

```diff
@@ -39,3 +45,125 @@ func LogIDFromPath(parent, path string) string {
 	logID := path[start:]
 	return strings.Replace(logID, "%2F", "/", -1)
 }
+
+var reCloudTraceContext = regexp.MustCompile(
+	// Matches on "TRACE_ID"
+	`([a-f\d]+)?` +
+		// Matches on "/SPAN_ID"
+		`(?:/([a-f\d]+))?` +
+		// Matches on ";0=TRACE_TRUE"
+		`(?:;o=(\d))?`)
+
+// DeconstructXCloudTraceContext extracts trace information from the trace
+// header.
+//
+// As per the format described at https://cloud.google.com/trace/docs/setup#force-trace
+//    "X-Cloud-Trace-Context: TRACE_ID/SPAN_ID;o=TRACE_TRUE"
+// for example:
+//    "X-Cloud-Trace-Context: 105445aa7843bc8bf206b120001000/1;o=1"
+//
+// We expect:
+//   * traceID (optional): 			"105445aa7843bc8bf206b120001000"
+//   * spanID (optional):       	"1"
+//   * traceSampled (optional): 	true
+func DeconstructXCloudTraceContext(s string) (traceID, spanID string, traceSampled bool) {
+	// As per the format described at https://cloud.google.com/trace/docs/setup#force-trace
+	//    "X-Cloud-Trace-Context: TRACE_ID/SPAN_ID;o=TRACE_TRUE"
+	// for example:
+	//    "X-Cloud-Trace-Context: 105445aa7843bc8bf206b120001000/1;o=1"
+	//
+	// We expect:
+	//   * traceID (optional): 			"105445aa7843bc8bf206b120001000"
```

Comment:
Tiny nit: the formatting/indents are off. Is it just how Github UI is rendering it?

Reply:
Whoops forgot to remove the inline comment when I added it to the function docs, good catch!